router/link_registry.go

```diff
@@ -29,6 +29,16 @@ type linkRegistryImpl struct {
 	ctrlCh channel.Channel
 }
 
+func (self *linkRegistryImpl) ControlChannel() channel.Channel {
+	// we may get link requests before the control channel is fully
+	// established. wait until it's set before we return. Will only
+	// happen right at startup
+	for self.ctrlCh == nil {
+		time.Sleep(30 * time.Millisecond)
```

Comment:
Shouldn't this be in a loop with a timeout? What is the machine is slow or bogged down?